qubes-gui-client/src/lib.rs

```diff
@@ -138,10 +138,21 @@ impl Client {
         Ok((s, conf))
     }
 
+    /// Try to reconnect.  If this fails, the agent is no longer usable; future
+    /// operations may panic.
+    pub fn reconnect(&mut self) -> io::Result<()> {
+        self.vchan.reconnect()
+    }
+
     /// Gets and clears the “did_reconnect” flag
     pub fn reconnected(&mut self) -> bool {
         self.vchan.reconnected()
     }
+
+    /// Returns true if a reconnection is needed.
+    pub fn needs_reconnect(&self) -> bool {
+        self.vchan.needs_reconnect()
+    }
```

Comment:
It'd expect some documentation that explains the expected use of these three functions, such as "once `needs_reconnect` returns `true`, one should call `reconnect` and then pump messages until `reconnected` returns `true` before further use of the `Client`".